packages/core/src/DdSdkReactNative.tsx

```diff
@@ -185,17 +190,19 @@ export class DdSdkReactNative {
     };
 
     /**
-     * Sets the global context (set of attributes) attached with all future Logs, Spans and RUM events.
+     * Adds a set of attributes to the global context attached with all future Logs, Spans and RUM events.
+     * To remove an attribute, set it to `undefined` in a call to `setAttributes`.
      * @param attributes: The global context attributes.
      * @returns a Promise.
      */
     // eslint-disable-next-line @typescript-eslint/ban-types
-    static setAttributes(attributes: object): Promise<void> {
+    static async setAttributes(attributes: Attributes): Promise<void> {
         InternalLog.log(
             `Setting attributes ${JSON.stringify(attributes)}`,
             SdkVerbosity.DEBUG
         );
-        return DdSdk.setAttributes(attributes);
+        await DdSdk.setAttributes(attributes);
+        AttributesSingleton.getInstance().setAttributes(attributes);
```

Comment:
Yes it is intended.
We used to `return DdSdk.setAttributes(attributes)` which itself returns a `Promise<void>` (promise resolving a value equal to `undefined`).
Now that we turned the function into an `async` one, and used `await`, it also returns a `Promise`, which in this case is a `Promise<void>` as we don't return anything.
This would be equivalent to `return;` or `return undefined` but the more common syntax is to not use a return statement.

Reply:
So using async makes an implicit return Promise, I guess this is knowledge I missed. 😅